### PR TITLE
[geo] Core work for GEO basic functionality

### PR DESCRIPTION
**Pull request by Omnione**
A bit big this one, take ya time going through it, still wip with debug stuff in it,
mainly to get feedback on systems.

no more spells yet, just indi-regen, geo-regen, indi-posion and geo-poison scripted,

done alot of backend systems to make things work correctly,
can now place a luopan using the control button to set a new location as an offset from the target,
luopan positions now set in core,
potency and geocolre effects are all worked out from the spell id not in spell scripts as with model of the loupan.
entrust targeting flags now in place ready for the ability to be setup corectly,
skillups now work with handbell if a handbell is equipped when casting a geo/indi spell.
Many more changes.

Please note, as stated this is a work in pregress and still being worked on and is not in a state to be used yet,
If you do, please do not make any issues related to geo as it will be ignored.

Any questions welcome.

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

File: src/map/packet_system.cpp [hunk 1]
```diff
@@ -735,6 +735,12 @@ void SmallPacket0x01A(map_session_data_t* const PSession, CCharEntity* const PCh
     // uint32 ID = data.ref<uint32>(0x04);
     uint16 TargID = data.ref<uint16>(0x08);
     uint8  action = data.ref<uint8>(0x0A);
+    position_t actionOffset =
+    {
+        data.ref<float>(0x10),
```
Comment (claywar) [inline src/map/packet_system.cpp:740]: General question, are we guaranteed that float on all architectures will always be 4 bytes?  If not, we might want to pull this in as something else and cast.

Comment (zach2good) [reply src/map/packet_system.cpp:740]: Yeah, that's a pretty safe assumption 👍

**Review by claywar — APPROVED** [review]

File: src/map/packet_system.cpp (hunk 1 above)
Comment (zach2good) [reply src/map/packet_system.cpp:740]: I think we can also rely on the endianness being correct, as we pull floats out of packets in other places

**Review by zach2good — APPROVED** [review]

File: src/map/lua/lua_baseentity.cpp [hunk 2]
```diff
@@ -10383,6 +10383,35 @@ bool CLuaBaseEntity::delLatent(uint16 condID, uint16 conditionValue, uint16 mID,
...
 }
 
+/************************************************************************
+ *  Function: getEquippedItemsMaxModValue()
+ *  Purpose : Returns the highest integer value of a specified Mod on all equiped items
+ *  Example : local maxValue = player:getEquippedItemsMaxModValue(xi.mod.GEOMANCY_BONUS)
+ *  Notes   :
+ ************************************************************************/
+
+int16 CLuaBaseEntity::getEquippedItemsMaxModValue(Mod modId)
```
Comment (TeoTwawki) [inline src/map/lua/lua_baseentity.cpp]: kinda long name. `getMaxGearMod()` ?

Comment (TeoTwawki) [reply src/map/lua/lua_baseentity.cpp]: related: do we mistakenly stack movement speed from gear? we must if we didn't already have a function to do this.

Comment (Omnione) [reply src/map/lua/lua_baseentity.cpp]: Yep, gets clamped is all

Comment (Omnione) [reply src/map/lua/lua_baseentity.cpp]: That’s a better name, I like getMaxGearMod, I’ll get that changed if others agree

**Comment by zach2good** [comment]
Yup

File: src/map/lua/lua_baseentity.cpp (hunk 2 above)
Comment (zach2good) [reply src/map/lua/lua_baseentity.cpp]: Sounds good to me

Comment (Omnione) [reply src/map/lua/lua_baseentity.cpp]: ok changed to getMaxGearMod()

**Review by TeoTwawki — APPROVED** [review]

**Pull request merged**